cli/src/main/java/com/devonfw/tools/ide/tool/ToolCommandlet.java

```diff
@@ -330,6 +330,52 @@ protected VersionIdentifier getInstalledVersion(Path toolPath) {
     }
   }
 
+  /**
+   * @return the installed edition of this tool or {@code null} if not installed.
+   */
+  public String getInstalledEdition() {
+
+    return getInstalledEdition(this.context.getSoftwarePath().resolve(getName()));
+  }
+
+  /**
+   * @param toolPath the installation {@link Path} where to find currently installed tool. The name of the parent
+   *        directory of the real path corresponding to the passed {@link Path path} must be the name of the edition.
+   * @return the installed edition of this tool or {@code null} if not installed.
+   */
+  public String getInstalledEdition(Path toolPath) {
+
+    if (!Files.isDirectory(toolPath)) {
+      this.context.debug("Tool {} not installed in {}", getName(), toolPath);
+      return null;
+    }
+    try {
+      String edition = toolPath.toRealPath().getParent().getFileName().toString();
+      if (!this.context.getUrls().getSortedEditions(getName()).contains(edition)) {
+        this.context.warning("The determined edition \"{}\" of tool {} is not among the editions provided by IDEasy",
+            edition, getName());
+      }
+      return edition;
```

Comment:
If the tool was installed via `devonfw-ide` this warning will be printed and the printed edition will be `software`.
IMHO this behavior does not really make sense.
In such case as fallback we could simply print the configured edition what is still better than printing `software`.

Reply:
Done. See #194